src/validate.rs

```diff
@@ -12,19 +12,43 @@
 
 use std::cmp::max;
 use std::collections::HashMap;
-use std::time::Instant;
 
 use crate::blockdir::Address;
 use crate::*;
 
-pub(crate) struct BlockLengths(pub(crate) HashMap<BlockHash, u64>);
+pub struct BlockLengths(pub HashMap<BlockHash, u64>);
 
 #[derive(Debug, Default)]
 pub struct ValidateOptions {
     /// Assume blocks that are present have the right content: don't read and hash them.
     pub skip_block_hashes: bool,
 }
 
+/// Band validation result.
+pub enum BandValidateResult {
```

Comment:
Maybe this should change to be just a regular 

    std::result::Result<(BlockLengths, ValidateStats), BandValidateError>

Reply:
I'm not a big fan of wrapping enums in enums but for continuity I changed it.

Reply:
I didn't mean an enum that wraps an Error, but rather that the error cases should simply be Errors. If the validate function hits a fatal problem it should return an `Err(Error)`